.github/workflows/subca-tests.yml

```diff
@@ -0,0 +1,194 @@
+name: Subordinate CA Tests
```

Comment:
Out of curiosity, why put this in a separate test? It'd be nicer to reuse the existing installation pipeline/... and stick this as another job inside there. Avoids another build job. The builds take a mostly constant amount of time, not like we're building smaller subsets of PKI for different jobs.

Reply:
The main issue is maintenance. There will be many more tests to be added, and that file will become really big. Using separate files the tests will be more organized, focused, easier to read, to troubleshoot, and to extend. Each test does spawn its own build job, but since they start in parallel they don't increase the overall execution time. If there's a way to share build artifacts that will be nice, but so far I don't see an easy way to do that.

Reply:
There's a limit to how much we can parallelize though -- I believe it is 7 jobs. If we get to where we're running > 7 pipelines, things could no longer execute in parallel. Right now, we're also at the limit: 3 jobs of IPA, 3 jobs of PKI install, and 3 jobs of this SubCA test means we won't ever execute them all in parallel, meaning we're waiting for some to finish. 

I'd prefer if we built a CI test suite that we executed inside the install/uninstall jobs (but also independently) for most scenarios. AFAICT, we don't actually test a SubCA in the install/uninstall scenario and this'd be a good addition IMO. (Along with light weight sub CA).

Ideally, CI could be simplified to execute a common CI script, that developers and QE could run independently on their own machines/VMs as well. Then we all don't have to dig through GitHub Actions' YAML files to see what was actually executed on the VM/container, in the event of failure. 

My 2c. Maybe we can work on this in another PR?

Reply:
OK, I've removed the new test for now, leaving just the code changes and the new doc.

This is the only thing that I found about the limit:
https://docs.github.com/en/free-pro-team@latest/actions/reference/usage-limits-billing-and-administration

IIUC the limit is for the total concurrent jobs per account, not per branch or PR. So if a lot of people are submitting PR at the same time, GH will only run 20 jobs at most in parallel, and the rest will wait in queue, but eventually they all will complete, just slower. So the limit is just describing the max number of job runners we get from GH, not the number of jobs we can add into the branch or PR. If we use a different GH plan we will get more runners, so the jobs will complete faster.

Also, even if we move the jobs into the same workflow, the number of jobs will not change, but the workflow file will become very big. I'm planning to add tests for all installation scenarios that we support:
https://github.com/dogtagpki/pki/wiki/Installation-Guide
so if they are all in the same workflow it might become unmanageable. We can discuss about this in a separate PR.